opencda/core/common/communication/serialize.py

```
@@ -111,127 +82,35 @@ def handle_artery_message(self, ego_id, id, module):
-
-                        # Process repeated type
-                        elif expected_label == FieldDescriptor.LABEL_REPEATED:
-                            if not isinstance(data, (list, tuple)):
-                                raise ValueError(f"[{entity_id}:{module_name}] Field '{key}' expected list or tuple, got {type(data)}")
-
-                            for i, item in enumerate(data):
-                                if not isinstance(item, expected_python_type):
-                                    raise ValueError(
-                                        f"[{entity_id}:{module_name}] Field '{key}' element at index {i} expected {expected_python_type.__name__}, got {type(item).__name__}"
-                                    )
-
-                            getattr(entity_message, key).extend(data)
-
-                        # Process scalar
-                        else:
-                            if not isinstance(data, expected_python_type):
-                                raise ValueError(
-                                    f"[{entity_id}:{module_name}] Field '{key}' expected type {expected_python_type.__name__}, got {type(data).__name__}"
-                                )
-                            setattr(entity_message, key, data)
-
-        return opencda_message.SerializeToString()
+            entity_message.auxillary = pickle.dumps(self.current_message_opencda[entity_id])
+
+        msg = proto_capi.Message(opencda=opencda_message)
+
+        return msg.SerializeToString()
 
     def make_artery_data(self, string) -> None:
-        artery_message = proto_capi.Artery_message()
-        artery_message.ParseFromString(string)
+        msg = proto_capi.Message()
+        msg.ParseFromString(string)
+
+        if msg.WhichOneof("message") == "artery":
+            artery_message = msg.artery
+        else:
+            logger.warning("Message does not contain ArteryMessage")
```

should raise?

---

Then we should raise an error there.

---

Done